### PR TITLE
Add branch information for Halide 12.

### PR DESCRIPTION
**Pull request by alexreinking**
Also replace LLVM-for-Halide function that was just a dictionary lookup with a global dictionary defined closer to the rest of the Halide branch information

### CODE REVIEW

**Comment by steven-johnson** [comment]
LGTM but lots of linter failures

**Comment by alexreinking** [comment]
> 
> 
> LGTM but lots of linter failures

Yup, fixing those.

**Comment by alexreinking** [comment]
@steven-johnson - take another look? I think the replacement check is appropriate (and maybe less yucky?).

File: master/master.cfg [hunk 1]
```diff
@@ -257,10 +257,7 @@ class BuilderType:
             assert self.purpose != Purpose.llvm_nightly
             assert self.halide_branch in HALIDE_BRANCHES, f'unknown branch {self.halide_branch}'
             assert (self.purpose == Purpose.halide_testbranch or  # if not testbranch...
-                    HALIDE_BRANCHES[self.halide_branch].version.major ==
-                    LLVM_BRANCHES[self.llvm_branch].version.major or
-                    (HALIDE_BRANCHES[self.halide_branch].version.major == 13 and
-                     LLVM_BRANCHES[self.llvm_branch].version.major == 14))  # TODO: yuck
+                    self.llvm_branch in LLVM_FOR_HALIDE[self.halide_branch])  # TODO: yuck
```
Comment (steven-johnson) [inline master/master.cfg:260]: ...do we still need `TODO: yuck`?

**Review by steven-johnson — APPROVED** [review]

File: master/master.cfg (hunk 1 above)
Comment (alexreinking) [reply master/master.cfg:260]: I guess we don't!

**Pull request merged**